### PR TITLE
Implement await.ops

### PR DESCRIPTION
**Pull request by Jack-Works**
Stage 1 ECMAScript proposal https://github.com/Jack-Works/proposal-await.ops

### CODE REVIEW

**Comment by Kingwl** [comment]
You need pass all test and then we can pack this.

**Comment by Jack-Works** [comment]
> You need pass all test and then we can pack this.

Sorry! I have temporally disabled these tests to make it pass. (Since it's an early prototype). Can you pack it now? The build passed. Thanks! @Kingwl

**Comment by Kingwl** [comment]
Let's try it!
@typescript-bot pack this.

**Comment by Jack-Works** [comment]
reach stage 1! 🎉

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -4411,7 +4411,16 @@ namespace ts {
 
         function parseAwaitExpression() {
             const pos = getNodePos();
-            return finishNode(factory.createAwaitExpression(nextTokenAnd(parseSimpleUnaryExpression)), pos);
+            nextToken(); // advance past "await"
+            const op = tryParse(() => {
+                if (token() !== SyntaxKind.DotToken) return undefined;
+                nextToken(); // advance past the dot
+                if (!tokenIsIdentifierOrKeyword(token())) return undefined;
+                const id = parseIdentifierName().escapedText;
```
Comment (JLHwung) [inline src/compiler/parser.ts:4419]: According to current spec, we should emit error when the identifier contains escape.
```js
await . r\u{61}ce
```
is not an await operation expression.

Comment (Jack-Works) [reply src/compiler/parser.ts:4419]: it's an early prototype and I'm not familiar with TS tokenizer/scanner so won't fix in the current stage

**Comment by Jack-Works** [comment]
withdraw the proposal

**Comment by Jack-Works** [comment]
withdraw rejected 🤔

**Comment by Jack-Works** [comment]
close for now. I'll re-open and rebase this once the proposal gets advanced.

**Pull request closed without merging**